src/Authorizer.ts

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios';
+import Cookies from 'js-cookie';
+import Permission from './Permission';
+import { StringKV } from './types';
+
+interface BaseResponse {
+    message: string;
+    data: any;
+}
+
+type Mode = "auto" | "cookies" | "manual"
+
+export class Authorizer {
+    private mode!: Mode;
+    private endpoint: string | undefined = undefined;
+    private user : string | undefined;
+    private permission = new Permission();
+    private cookieKey : string | undefined = undefined;
+
+    /**
+     * 
+     * @param mode "auto", "cookies" or "manual"
+     * "auto": Specify the casbin server endpoint, and Casbin.js will load permission from it when the identity changes
+     * "cookies": Casbin.js load the permission data from the cookie "casbin_permission" or the specified cookie key.
+     * "manual": Load the permission mannually with "setPermission"
+     * @param args.endpoint Casbin service endpoint, required when mode == "auto"
+     * @param args.cookieKey The cookie key when loading permission, activated when mode == "cookies"
+     */
+    constructor(mode: Mode, args: {endpoint?: string, cookieKey?: string} = {}) {
```

Comment:
Here are set default value for mode is better: `constructor(mode: Mode = 'manual', args: {endpoint?: string, cookieKey?: string} = {}) `